src/System.Runtime.Serialization.Formatters/tests/TypeSerializableValue.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Linq;
+
+namespace System.Runtime.Serialization.Formatters.Tests
+{
+    public class TypeSerializableValue
+    {
+        public TypeSerializableValue(string base64Blob, TargetFrameworkMoniker platform)
+        {
+            Base64Blob = base64Blob;
+            Platform = platform;
+        }
+
+        public string Base64Blob { get; }
+
+        public TargetFrameworkMoniker Platform { get; }
+
+        public static int GetPlatformIndex(TypeSerializableValue[] blobs)
+        {
+            var blobList = blobs.ToList();
+
+            // .NET Framework
+            if (PlatformDetection.IsFullFramework)
+            {
+                if (!PlatformDetection.IsNetfx471OrNewer)
+                {
+                    // Check if a specialized blob for >=net471 is present and return if found.
+                    int index = blobList.FindIndex(b => b.Platform == TargetFrameworkMoniker.netfx);
+                    if (index >= 0)
+                        return index;
+                }
+
+                // If no newer blob for >=net471 is present use existing one. 
+                // If no netfx blob is present then -1 will be returned.
+                return blobList.FindIndex(b => b.Platform == TargetFrameworkMoniker.netfx471);
+            }
+
+            // .NET Core
+            return PlatformDetection.IsNetCore ? blobList.FindIndex(b => b.Platform == TargetFrameworkMoniker.netcoreapp21) : -1;
```

Comment:
If I'm running on `netcoreapp21` (master currently) I will never get an index back for blobs that haven't changed with `netcoreapp21`. I believe you should first check for `netcoreapp21` and if index is < 0 then check for `netcoreapp`.